.github/workflows/pkgci_build_packages.yml

```
@@ -0,0 +1,71 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Build Packages
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  build_packages:
+    runs-on: ubuntu-24.04
+    env:
+      CACHE_DIR: ${{ github.workspace }}/.shark-ai-cache
+      OUTPUT_DIR: "${{ github.workspace }}/bindist"
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+    - name: Setup Python
+      uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+      with:
+        python-version: 3.12
+
+    # Setup metadata.
+    - name: Install Python packages
+      run: pip install packaging
+    - name: Generate dev package versions
+      id: version_rc
+      run: |
+        version_suffix=".dev+${{ github.sha }}"
```

```suggestion
        version_suffix=".dev0+${{ github.sha }}"
```

The version is normalized when building the packages and we might want to be explicit here:
```
shark_ai-3.1.0.dev0+20f1ecb69a24049675377cb4b41d53d599cda5dd-py3-none-any.whl
sharktank-3.1.0.dev0+20f1ecb69a24049675377cb4b41d53d599cda5dd-py3-none-any.whl
shortfin-3.1.0.dev0+20f1ecb69a24049675377cb4b41d53d599cda5dd-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```